shared/video-sdk/authentication-workflow/project-implementation/cpp.mdx

```diff
@@ -100,33 +119,27 @@
     with:
 
     ```cpp
-    if (token == "")
-    {
-        // Fetch new token 
-        token = fetchToken(serverUrl, channelName, tokenRole, uid, expireTime);
-        if (token == "")
-        {
-            AfxMessageBox(L"Invalid Token.");
-            return;
-        }
-    }
-    if (0 == agoraEngine->joinChannel(token.c_str(), channelName.c_str(), NULL, uid))
-    {
-        isJoin = true;
-    }
+    CString channel;
+	GetDlgItemText(IDC_EDIT1, channel);
+	channelName = std::string(CT2CA(channel));
+	if(channel == "")
+	{
+		AfxMessageBox(L"Please write a channel name in the text box");
+		return;
+	}
+	else
+	{
+		// Fetch new token 
+		token = fetchToken(serverUrl, channelName, tokenRole, uid, expireTime);
```

Comment:
Actually with this change it will miss the case  when token is not generated because of return status of chttps reuest  NOT_OK.. in that case token will be blank..